etc/picongpu/hypnos-hzdr/k80_restart.tpl

```diff
@@ -99,33 +99,62 @@ do
         file=""
     fi
 done
+export file
 
 #this sed call extracts the final simulation step from the cfg (assuming a standard cfg)
-finalStep=`echo !TBG_programParams | sed 's/.*-s[[:blank:]]\+\([0-9]\+[^\s]\).*/\1/'`
+export finalStep=`echo !TBG_programParams | sed 's/.*-s[[:blank:]]\+\([0-9]\+[^\s]\).*/\1/'`
 echo "final step      = " $finalStep | tee -a output
 #this sed call extracts the -s and --checkpoint flags
-programParams=`echo !TBG_programParams | sed 's/-s[[:blank:]]\+[0-9]\+[^\s]//g' | sed 's/--checkpoint\.period[[:blank:]]\+[0-9]\+[^\s]//g'`
+programParams=`echo !TBG_programParams | sed 's/-s[[:blank:]]\+[0-9]\+[^\s]//g' | sed 's/--checkpoint\.period[[:blank:]]\+[0-9,:,\,]\+[^\s]//g'`
 #extract restart period
-restartPeriod=`echo !TBG_programParams | sed 's/.*--checkpoint\.period[[:blank:]]\+\([0-9]\+[^\s]\).*/\1/'`
+export restartPeriod=`echo !TBG_programParams | sed 's/.*--checkpoint\.period[[:blank:]]\+\([0-9,:,\,]\+[^\s]\).*/\1/'`
 echo  "restart period = " $restartPeriod | tee -a output
 
-if [ "" != "$file" ]
-then
-    cptimestep=`basename $file | sed 's/checkpoint_//g' | sed 's/.'$fileEnding'//g'`
-    echo "start time      = " $cptimestep | tee -a output
-
-    endTime="$(($cptimestep + $restartPeriod ))"
-    echo "end time        = " $endTime | tee -a output
-
-    stepSetup=$(echo -s $endTime "--checkpoint.restart --checkpoint.restart.step" $cptimestep "--checkpoint.period" $restartPeriod )
-else
-    echo "no checkpoint found" | tee -a output
-    endTime=$restartPeriod
-    stepSetup=$(echo " -s " $endTime "--checkpoint.period" $restartPeriod )
-fi
+# ******************************************* #
+# need some magick, if the restart period is in new notation with the ':' and ','
+
+which python3 >&2
+python3 --version >&2 
+read -d "\n" endTime stepSetup <<< `python3 -c "
+import os, sys
+# create list of timesteps to be dumped
+steps = []
+restartPeriod = os.environ['restartPeriod']
+intervals = restartPeriod.split(',')
+for interval in intervals:
+    parts = [None if not part else int(part) for part in interval.split(':')]
+    if len(parts)==1:
+        parts = [None, None, parts[0]]
+    elif len(parts)==2:
+        parts = [parts[0], None if parts[1] is None else parts[1]+1, None]
+    elif len(parts)==3:
+        parts = [parts[0], None if parts[1] is None else parts[1]+1, parts[2]]
+
+    steps += range(int(os.environ['finalStep'])+1)[slice(*parts)]
+
+steps.append(int(os.environ['finalStep']))
+print('steps-Liste: %s' % steps, file=sys.stderr)
+# return list and next bigger timestep as endpoint for next run
+if not os.environ['file']:
+    next_step = min(steps)
+else:
+    current_step = int(os.path.splitext(os.path.basename('$file'))[0].strip('checkpoint_'))
```

Comment:
just a remark: this only works if `--checkpoint.restart.file` hasn't been changed

Reply:
That is true for the current version as well. Thanks for pointing this out.